### PR TITLE
Corrected batch norm momentum, init of models, data aug strategies and coupled weight decay application for TF

### PR DESCRIPTION
**Pull request by zaccharieramzi**
These corrections are needed to have exactly the same model even if they are not catchable with the current unit test.

They also do not change the final model performance.

EDIT
-----
Now only the init of models is not catchable by unit tests.
Moreover, it was really the coupled weight decay application that allowed to reach PyTorch's perf.

Unit tests are not passing in eval mode because of the difference in variance estimation for the inference between PyTorch and TensorFlow (see https://github.com/pytorch/pytorch/issues/77427#issuecomment-1126737756).
Regarding the adam test, and the sgd with momentum alone, I am not sure yet what the cause is.

### CODE REVIEW

**Comment by zaccharieramzi** [comment]
For now 2 types of unit tests are passed:
- unit test in eval mode, where there is a discrepancy for PyTorch
- unit test for Adam, which leaves #43 open
